pydocxs3upload/test/utils.py

```diff
@@ -134,15 +134,20 @@ def request_callback(request):
             rbody = request.body
 
             _, pdict = cgi.parse_header(request.headers['Content-Type'])
+            pdict['boundary'] = bytes(pdict['boundary'], 'utf-8')
             data = cgi.parse_multipart(BytesIO(rbody), pdict)
 
             # TODO find a better way to take filename
-            filename = re.search(r'filename="(.+?)"', rbody).group(1)
+            filename = re.search(r'filename="(.+?)"', str(rbody)).group(1)
 
-            key_name = data['key'][0].replace('${filename}', filename)
-            bucket_name = _get_bucket_name(data['policy'][0])
+            key_name_string = data['key'][0].decode("utf-8")
+            policy = data['policy'][0].decode("utf-8")
```

Comment:
decode to strings for use below

Reply:
Is the decoding for Python3? Or for good practice?

Reply:
Most of these conversions are because python 2.x's cgi.py used strings for it's arguments. 3.x uses bytes and returns bytes. I'm decoding to strings on these two lines because the code that uses these two vars wants strings.